### ISSUE TITLE
Support for Python 3

### ISSUE THREAD

**Issue by GoogleCodeExporter** [davidlbaird/pyfakefs#14]
```
Test pyfakefs in Python 3.2 and 3.3 and create changes in a branch to be 
reviewed, and merged to trunk.
```

Original issue reported on code.google.com by `dba...@google.com` on 1 Mar 2013 at 11:40

**Comment by GoogleCodeExporter** [comment]
Original comment by `dba...@google.com` on 1 Mar 2013 at 11:45

**Comment by GoogleCodeExporter** [comment]
```
I can't create a branch in this tree, so I made a clone: 
https://code.google.com/r/kevin-pyfakefs-python3/source/list?name=python3

In terms of doing a review, there are some changes I'll specifically highlight:.

Two types were deprecated in python3. I deal with both the same way - though 
the later one is in the actual fakefs code while the former is in the code that 
tests it. Dealing with the latter one at the module level would be more 
efficient.

  * basestring deprecation https://code.google.com/r/kevin-pyfakefs-python3/source/detail?r=bd311af0de28449083c8afea312311a733edb8cb&name=python3
  * long deprecation https://code.google.com/r/kevin-pyfakefs-python3/source/detail?r=e7944ed4648e9fb4bb49f7511ce44a4090f6d995&name=python3

Annoyingly I didn't fix exception issues in a single CL. Sorry. I'd know how to 
fix that in git, but not in hg. The revisions that fix exception handling are 
here:

  * Use the correct IOError members https://code.google.com/r/kevin-pyfakefs-python3/source/detail?r=5e697cc1c4d44642365eb0d34221422cefbdeca0&name=python3
  * Scoping fix https://code.google.com/r/kevin-pyfakefs-python3/source/detail?r=0e413f9ebaf025eebb2f11b5280d460406077aa7&name=python3
  * Initial exception handling fix https://code.google.com/r/kevin-pyfakefs-python3/source/detail?r=f10d19a4274ab14ec8eb18c933dfefa467d9a1c1&name=python3

Hope these are clear.
```

Original comment by `kevin@ie.suberic.net` on 2 Mar 2013 at 11:12

**Comment by GoogleCodeExporter** [comment]
Original comment by `dba...@google.com` on 21 May 2013 at 12:01
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Issue 13 has been merged into this issue.
```

Original comment by `dba...@google.com` on 21 May 2013 at 8:07

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 4300e25d1a3c.
```

Original comment by `dba...@google.com` on 21 May 2013 at 9:39
- Changed state: **Fixed**
